File: datasets/C++/Deepseek/Algorithm/easy5.cpp
#include <iostream>
#include <vector>

using namespace std;

string gameOfStones(int n) {
    vector<bool> dp(101, false);
    dp[0] = false;
    dp[1] = false;
    for (int i = 2; i <= 100; ++i) {
        if (i >= 2 && !dp[i - 2]) {
            dp[i] = true;
        } else if (i >= 3 && !dp[i - 3]) {
            dp[i] = true;
        } else if (i >= 5 && !dp[i - 5]) {
            dp[i] = true;
        } else {
            dp[i] = false;
        }
    }
    return dp[n] ? "First" : "Second";
}

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        cout << gameOfStones(n) << endl;
    }
    return 0;
}

File: datasets/Python/Chatgpt/Algorithm/easy5.py
def gameOfStones(n):
    # Using dynamic programming
    dp = [False] * (n + 1)  # False means losing position

    for i in range(1, n + 1):
        if i >= 2 and not dp[i - 2]:
            dp[i] = True
        elif i >= 3 and not dp[i - 3]:
            dp[i] = True
        elif i >= 5 and not dp[i - 5]:
            dp[i] = True
        # else it remains False

    return "First" if dp[n] else "Second"

# Input
t = int(input("Enter number of test cases: "))
for _ in range(t):
    n = int(input("Enter number of stones: "))
    print(gameOfStones(n))
